src/cfnlint/rules/resources/HardCodedArnProperties.py

```diff
@@ -0,0 +1,34 @@
+"""
+Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+import re
+from cfnlint.rules import CloudFormationLintRule
+from cfnlint.rules import RuleMatch
+
+
+class HardCodedArnProperties(CloudFormationLintRule):
+    """Checks Resources if ARNs use correctly placed Pseudo Parameters instead of hardcoded Partition, Region, and Account Number"""
+    id = 'W3042'

```

Comment:
so I've been writing a warning-level rule for hardcoded partitions because partitions cannot communicate with each other but there are valid reasons for using other regions and account IDs

maybe this rule should be informational-level?

Reply:
> Given the same case it wouldn't make sense to throw a warning on a hardcoded Partition either

partitions cannot communicate with each other, so I don't think hardcoded partitions are ever necessary, whereas cross-account and cross-region communication is a valid use-case, so I think hardcoded accounts and regions should probably only be informational. [Separate from this rule specifically, I would like informational rules to start showing by default without affecting the exit code](https://github.com/aws-cloudformation/cfn-python-lint/issues/1586#issuecomment-642984414) and only affect the exit code if opted into `--include-checks I`